services/src/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/PathHelper.groovy

```diff
@@ -68,8 +67,16 @@ class PathHelper {
             throw new Exception("Unable to make some path more preferable because both paths use same ISLs")
         }
         log.debug "ISL to avoid: $islToAvoid"
-        northbound.updateLinkProps([islUtils.toLinkProps(islToAvoid, ["cost": NOT_PREFERABLE_COST]),
-                                    islUtils.toLinkProps(islToAvoid.reversed, ["cost": NOT_PREFERABLE_COST])])
+        def totalCostOfMorePrefPath = morePreferableIsls.collect {
+            northbound.getLink(it)
+        }.cost.sum()
+        def totalCostOfLessPrefPath = getInvolvedIsls(lessPreferablePath).collect {
+            northbound.getLink(it)
+        }.cost.sum()
+        if(totalCostOfLessPrefPath <= totalCostOfMorePrefPath) {
```

Comment:
the `islToAvoid`-related code should be moved inside this if

Reply:
fixed

Reply:
I don't see the change. it should be 
```
if(totalCostOfLessPrefPath <= totalCostOfMorePrefPath) {
         def islToAvoid = getInvolvedIsls(lessPreferablePath).find {
            !morePreferableIsls.contains(it) && !morePreferableIsls.contains(it.reversed)
        }
        if (!islToAvoid) {
            throw new Exception("Unable to make some path more preferable because both paths use same ISLs")
        }
    ....
     .....
}
```
Also, javadoc should be updated according to the new behavior. Also it sohuld be mentioned that 'null' can be returned if we didn't change any isls.